### PR TITLE
Support system/pseudo column as version field (ORA_ROWSCN)

### PR DESCRIPTION
**Pull request by radovanradic**

### CODE REVIEW

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+@MappedEntity
+class Item {
+    @Id
+    @GeneratedValue
+    Long id
+
+    String name
+
+    Double price
+
```
Comment (radovanradic) [inline data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy:55]: The assumption is if we have `@Version` and `@GeneratedValue` we don't insert/update values or use field in the create table script. Not sure if this is reasonable, but was simplest approach. Maybe there would be cases when such field is not actual system field, then we might need to introduce something like `@SystemField` annotation.

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    ApplicationContext context = ApplicationContext.run(properties)
+
+    @Memoized
+    ItemRepository getItemRepository() {
+        return context.getBean(ItemRepository)
+    }
+
+    void "test optimistic lock with ORA_ROWSCN"() {
+        when:
+        def item = itemRepository.save(new Item(name: "Sneakers", price: 119.99))
```
Comment (dstepanov) [inline data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy:30]: Please try fetch all the generated values

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy (hunk 1 above)
Comment (dstepanov) [reply data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy:55]: Initially I thought that the version is always marked as generated value, this looks fine

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy (hunk 2 above)
Comment (radovanradic) [reply data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/OracleXEOraRowScnSpec.groovy:30]: Looks like retrieving pseudo columns does not work, when tried to return ["id", "ora_rowscn"] as generated keys got this error`java.sql.SQLException: Invalid column name: ora_rowscn` while ["id", "name"] worked ok.

**Review by dstepanov — APPROVED** [review]

**Pull request merged**